### ISSUE TITLE
Difference in Test Error on BN models

### ISSUE THREAD

**Issue by rdev1711** [timgaripov/dnn-mode-connectivity#8]
I have tried training both VGG16BN and VGG19BN Models, as well as generating the curve between model checkpoints. I find that the test error of the base model is different than that of the curve model at t=0 (it is my understanding that at t=0 and t=1, the curve model should be identical to each of the checkpoints. Is that correct?

When I train the VGG16 and VGG19 models (without batch normalization), the test errors are the same. I think this may point to some error in the dataset loading. Is there something different being done with the batch normalization models from the standard definitions?

Thanks.

**Comment by izmailovpavel** [comment]
Hi @rdev1711. The most likely explanation is that for BatchNorm models there is a difference between train and eval modes, and in particular you need to update the BatchNorm statistics for the curve model samples with the `update_bn` [function](https://github.com/timgaripov/dnn-mode-connectivity/blob/77b67db68f3fe20eefddd0ec73275250221ac61f/utils.py#L149).
See e.g. the evaluation script [here](https://github.com/timgaripov/dnn-mode-connectivity/blob/77b67db68f3fe20eefddd0ec73275250221ac61f/eval_curve.py#L103).
